crates/services/gas_price_service/src/v1/da_source_adapter/block_committer_costs.rs

```diff
@@ -6,57 +9,334 @@ use crate::v1::da_source_adapter::{
     DaBlockCosts,
 };
 use anyhow::anyhow;
-use reqwest::Url;
+use fuel_core_types::blockchain::primitives::DaBlockHeight;
 use serde::{
     Deserialize,
     Serialize,
 };
 
-/// This struct is used to denote the block committer da gas price source,,
+#[async_trait::async_trait]
+trait BlockCommitterApi: Send + Sync {
+    // Used on first run to get the latest costs and seqno
+    async fn get_latest_costs(&self) -> DaBlockCostsResult<Option<RawDaBlockCosts>>;
+    // Used to get the costs for a specific seqno
+    async fn get_costs_by_seqno(
+        &self,
+        number: u64,
+    ) -> DaBlockCostsResult<Option<RawDaBlockCosts>>;
+    async fn get_bundles_by_range(
```

Comment:
I think adding `costs` somewhere to the name would make it easier to relate this function to the other two. 
E.g. `get_cost_bundles_by_range`

Reply:
addressed in 552d22479a